src/components/LoginMenu.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <v-menu
+    v-model="menu"
+    offset-x
+  >
+    <template v-slot:activator="{ on }">
+      <v-btn
+        icon
+        v-on="on"
+      >
+        <v-avatar
+          color="grey lighten-4"
+          size="36px"
+        >
+          <span v-if="userInfo">{{ userInitials }}</span>
+          <v-icon
+            v-else
+            color="grey"
+          >
+            account_circle
+          </v-icon>
+        </v-avatar>
+      </v-btn>
+    </template>
+
+    <v-card>
+      <v-list>
+        <v-list-item class="sign-in-item">
+          <v-list-item-action>
+            <v-btn
+              v-if="userInfo"
+              dark
+              @click="logout"
+            >
+              Log out
+            </v-btn>
+
+            <v-btn
+              v-else
+              class="google-sign-in"
+              dark
+              :ripple="false"
+              :href="loginLink"
+            >
+              <span class="google-logo">
+                <img
+                  alt="Google"
+                  src="../assets/google_badge.svg"
+                >
+              </span>
+              <span class="button-text">
+                Sign in with Google
+              </span>
+            </v-btn>
+          </v-list-item-action>
+        </v-list-item>
+      </v-list>
+    </v-card>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import { host } from '@/environment';
+import store from '@/store';
+import { UserInfo } from '@/types';
+
+export default {
+  data: () => ({
+    menu: false,
+    location: '',
+  }),
+
+  computed: {
+    userInfo: (): UserInfo | null => store.state.userInfo,
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    loginLink(this: any): string {
+      const {
+        location,
+      } = this;
+
+      return `${host}/user/oauth/google/login?return_url=${location}`;
+    },
+
+    userInitials(): string {
+      const {
+        userInfo,
+      } = this;
+
+      if (userInfo !== null) {
+        return `${(userInfo as unknown as UserInfo).given_name[0]}${(userInfo as unknown as UserInfo).family_name[0]}`;
```

Comment:
Actually, I don't think the null-guard is needed anymore. When you cast the type of `userInfo` to `UserInfo` rather than `UserInfo | null`, the need for the guard evaporates.

I'll try this out in a fresh commit.

Reply:
Did you ever try this out? I've got this feeling that because of the `as unknown` part of that type cast, it wouldn't actually catch it, but that's just a feeling.

Reply:
I'm about to try it now. I suppose we should be casting it to `UserInfo | null`, and then include a return value for the `null` case.